test/crypto-service/test_accountUpdateTransaction.js

```diff
@@ -0,0 +1,1041 @@
+
+describe("AccountUpdateTransaction", function () {
+  // Tests should not take longer than 30 seconds to fully execute.
+  this.timeout(30000);
+
+  // An account is created for each test. These hold the information for that account.
+  let accountPrivateKey, accountId;
+  
+  beforeEach(async function () {
+    // Initialize the network and operator.
+    await setOperator(process.env.OPERATOR_ACCOUNT_ID, process.env.OPERATOR_ACCOUNT_PRIVATE_KEY);
+
+    // Generate a private key.
+    let response = await JSONRPCRequest("generateKey", {
+      type: "ed25519PrivateKey"
+    });
+    if (response.status === "NOT_IMPLEMENTED") this.skip();
+    accountPrivateKey = response.key;
+
+    // Create an account using the generated private key.
+    response = await JSONRPCRequest("createAccount", {
+      key: accountPrivateKey
+    });
+    if (response.status === "NOT_IMPLEMENTED") this.skip();
+    accountId = response.accountId;
+  });
+  afterEach(async function () {
+    await JSONRPCRequest("reset");
+  });
+
+  describe("AccountId", async function () {
+    it("(#1) Updates an account with no updates", async function () {
+      // Attempt to update the account.
+      const response = await JSONRPCRequest("updateAccount", {
+        accountId: accountId,
+        commonTransactionParams: {
+          signers: [
+            accountPrivateKey
+          ]
+        }
+      });
+      if (response.status === "NOT_IMPLEMENTED") this.skip();
+
+      // Account info should remain the same
+      let mirrorNodeData = await mirrorNodeClient.getAccountData(accountId);
+      let consensusNodeData = await consensusInfoClient.getAccountInfo(accountId);
+      expect(accountId).to.be.equal(mirrorNodeData.accounts[0].account);
+      expect(accountId).to.be.equal(consensusNodeData.accountId.toString());
+    });
+
+    it("(#2) Updates an account with no updates without signing with the account's private key", async function () {
+      try {
+        // Attempt to update the account without signing with the account's private key. The network should respond with an INVALID_SIGNATURE status.
+        const response = await JSONRPCRequest("updateAccount", {
+          accountId: accountId
+        });
+        if (response.status === "NOT_IMPLEMENTED") this.skip();
+      } catch(err) {
+        assert.equal(err.data.status, "INVALID_SIGNATURE");
+        return;
+      }
+
+      // The test failed, no error was thrown.
+      assert.fail("Should throw an error");
+    });
+
+    it("(#3) Updates an account with no account ID", async function () {
+      try {
+        // Attempt to update the account without providing the account ID. The network should respond with an INVALID_ACCOUNT_ID status.
+        const response = await JSONRPCRequest("updateAccount", {});
+        if (response.status === "NOT_IMPLEMENTED") this.skip();
+      } catch(err) {
+        assert.equal(err.data.status, "INVALID_ACCOUNT_ID");
```

Comment:
I am getting `ACCOUNT_ID_DOES_NOT_EXIST` error instead of `INVALID_ACCOUNT_ID`, 
this can be seen in services as well
```java
        validateTruePreCheck(op.hasAccountIDToUpdate(), ACCOUNT_ID_DOES_NOT_EXIST);

```

Reply:
Yep, this was actually a bug on the C++ SDK side. Fixed